src/components/Launch/LaunchWorkflowForm/useFormInputsState.ts

```diff
@@ -0,0 +1,83 @@
+import { useDebouncedValue } from 'components/hooks/useDebouncedValue';
+import { Core } from 'flyteidl';
+import { useEffect, useState } from 'react';
+import { validateInput } from './inputHelpers/inputHelpers';
+import { InputProps, InputValue, ParsedInput } from './types';
+import { convertFormInputsToLiterals } from './utils';
+
+const debounceDelay = 500;
+
+interface FormInputState extends InputProps {
+    validate(): boolean;
+}
+
+interface FormInputsState {
+    inputs: InputProps[];
+    getValues(): Record<string, Core.ILiteral>;
+    validate(): boolean;
+}
+
+function useFormInputState(parsedInput: ParsedInput): FormInputState {
+    const [value, setValue] = useState<InputValue>();
+    const [error, setError] = useState<string>();
+
+    const validationValue = useDebouncedValue(value, debounceDelay);
+
+    const validate = () => {
+        const { typeDefinition } = parsedInput;
+        try {
+            validateInput({ name, typeDefinition, value });
+            setError(undefined);
+            return true;
+        } catch (validationError) {
+            setError(`${validationError}`);
+            return false;
+        }
+    };
+
+    useEffect(() => {
+        validate();
+    }, [validationValue]);
+
+    const onChange = (value: InputValue) => {
+        setValue(value);
+    };
+
+    return {
+        ...parsedInput,
+        error,
+        onChange,
+        validate,
+        value,
+        helperText: parsedInput.description
+    };
+}
+
+/** Manages the state (value, error, validation) for a list of `ParsedInput` values.
+ * NOTE: The input value for this hook is used to generate sub-hooks.
```

Comment:
Good to have this warning here -- I was trying to think of a way to enforce it on the code side, but you probably don't want any reference  to the mount state of the component